lib/internal/streams/writable.js

```diff
@@ -263,6 +265,8 @@ function Writable(options) {
 
     if (typeof options.construct === 'function')
       this._construct = options.construct;
+    if (options.signal)
```

Comment:
Do we need a more specific type check?

Reply:
Otherwise I think we might need to consider semver major.

Reply:
addAbortSignal does a more specific type check already

Reply:
Does that error or ignore?

Reply:
It errors with `ERR_INVALID_ARG_TYPE`

Reply:
That might be a problem. If existing code provides a signal that is not intended for stream. Maybe ignore is better?

Reply:
That's a good point 👍 I will ignore the argument.